src/queries/full-text-queries/query-string-query.js

```diff
@@ -303,6 +309,42 @@ class QueryStringQuery extends QueryStringQueryBase {
         this._queryOpts.escape = enable;
         return this;
     }
+
+    /**
+     * Sets the type of query string query. Valid values are:
+     * - `best_fields` - (default) Finds documents which match any field,
+     * but uses the `_score` from the best field.
+     *
+     * - `most_fields` - Finds documents which match any field and combines
+     * the `_score` from each field.
+     *
+     * - `cross_fields` - Treats fields with the same `analyzer` as though
+     * they were one big field. Looks for each word in *any* field
+     *
+     * - `phrase` - Runs a `match_phrase` query on each field and combines
+     * the `_score` from each field.
+     *
+     * - `phrase_prefix` - Runs a `match_phrase_prefix` query on each field
+     * and combines the `_score` from each field.
+     *
+     * - `bool_prefix` - Creates a match_bool_prefix query on each field and
+     * combines the _score from each field.
+     *
+     *
+     * @param {string} type Can be one of `best_fields`, `most_fields`,
+     * `cross_fields`, `phrase`, `phrase_prefix` and `bool_prefix`. Default is
+     * `best_fields`.
+     * @returns {QueryStringQuery} returns `this` so that calls can be chained.
+     */
+    type(type) {
```

Comment:
Could you please update the type definition as well - https://github.com/sudo-suhas/elastic-builder/blob/v2.16.0/src/index.d.ts#L996-L1078

Reply:
Should we just have an enum for these type values? They are duplicated in the multi match query. Also in the .js files these values are duplicated. How do you want to handle this?

Reply:
Could you clarify what you mean by an enum? The source is not using Typescript yet.

> They are duplicated in the multi-match query.

Yes, we could re-use `MULTI_MATCH_TYPE` defined in the `src/core/consts.js` file.

Reply:
I've been a bit busy but will get to this soon. I have also found 1 more area that I want to improve and will just include it with this PR since as well. 

I'll get some changes pushed up next week to get this (hopefully) approved and released.